packages/odata-common/src/request-builder/action-function-import-request-builder-base.ts

```diff
@@ -15,14 +17,26 @@ export abstract class ActionFunctionImportRequestBuilderBase<
    * Base class for function  and actions imports.
    * @param responseTransformer - Transformation function for the response.
    * @param requestConfig - Request config for a action or function import.
+   * @param batchReference - Identifier for the batch request.
    */
   protected constructor(
     readonly responseTransformer: (data: any) => ReturnT,
-    requestConfig: RequestConfigT
+    requestConfig: RequestConfigT,
+    private batchReference: BatchReference = { id: uuid() }
```

Comment:
[q] Could you add the `batchReference` property to the `MethodRequestBuilder` class which is the parent for all the request builders? Same is true for the `getBatchReference` and `setBatchId` so that you do not have to duplicate the code multiple times?

Reply:
I got the same idea and Marika planned that too. When that is placed in `MethodRequestBuilder`, we can reduce the duplicated codes across the `RequestBuilderBase`s, and also the complicated union type and type guard in the serializer can be removed because just `MethodRequestBuilder` is enough in the case.It's simple and looks better. 
However, Junjie suggested that all  `RequestBuilderBase`s don't need the `batchReference`, the getter, and the setter. `MethodRequestBuilder ` may be too broad for that.
@marikaner has already aware of these pros and cons, she will add comments.

Reply:
ok, sure. No matter what you decide I would like to define a interface `ReferencableInBatch` or something which defines the getter,setter methods and is added to the request builders under implements so that the methods are ways the same. Perhaps @marikaner can also comment if the init should be done in constructor or lazy in the get method.

Reply:
Yes, I discussed with Shoji regarding the abstraction of `batchReference`.
There is a tradeoff as not all the request builders need it.
Unfortunately, Marika cannot provide feedback for now.
@FrankEssenberger , I checked the changes of this PR, and feel we can wait for Marika.
Even we create a v3 branch and this amount of files can still be handled and merged to two branches.
Let me know, if you want to merge before the v3 branch.

Reply:
I agree with Frank

Reply:
perhaps TS supports default implementation for interface finally, then we could safe even the coding.